lib/commands/publish.js

```diff
@@ -96,6 +96,10 @@ class Publish extends BaseCommand {
     }
 
     const resolved = npa.resolve(manifest.name, manifest.version)
+
+    // make sure tag is valid, this will throw if invalid
+    npa(`${manifest.name}@${defaultTag}`)
```

Comment:
Note: this matches https://github.com/npm/cli/blob/8f7fd4f2ec37956780c9f3b1cd748bfebdfb863c/lib/commands/dist-tag.js#L92-L93